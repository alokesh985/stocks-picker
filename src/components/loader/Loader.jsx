import React, { memo } from "react";
import PropTypes from 'prop-types';

// Styles
import styles from './loader.module.scss';

// Loader svg
import loader from "./assets/svg/loader.svg"

const Loader = ({ className }) => <img className={`${styles.loader} ${className}`} src={loader} alt="loader"/>;

Loader.propTypes = {
  className: PropTypes.string,
};

Loader.defaultProps = {
  className: '',
}

export default memo(Loader);