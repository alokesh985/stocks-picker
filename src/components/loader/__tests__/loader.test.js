// Utils
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Components
import Loader from "../Loader";

describe("Testing the Loader component", () => {
  test("Should match snapshot", () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should render 1 loader", () => {
    render(<Loader />);
    const loaderImage = screen.getAllByAltText("loader");
    expect(loaderImage.length).toBe(1);
  })
});
