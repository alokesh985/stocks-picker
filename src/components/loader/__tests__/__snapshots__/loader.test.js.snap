// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Loader component Should match snapshot 1`] = `
<img
  alt="loader"
  className="loader "
  src="loader.svg"
/>
`;
